packages/devtools_app/lib/src/timeline/timeline_controller.dart

```diff
@@ -116,70 +124,114 @@ class TimelineController {
     _cpuProfileTransformer.processData(cpuProfileData);
   }
 
-  void loadOfflineData(OfflineTimelineData offlineData) {
-    // TODO(kenz): loading offline data should respect user's current timeline
-    // mode (frameBased vs full). It should also support toggling modes.
-    frameBasedTimeline.data = offlineData.copy();
-    offlineTimelineData = offlineData.copy();
-
-    final traceEvents =
-        offlineData.traceEvents.map((trace) => TraceEvent(trace)).toList();
+  void loadOfflineData(TimelineData offlineData) {
+    final traceEvents = offlineData.traceEvents
+        .map((trace) => TraceEventWrapper(
```

Comment:
nit: switch this to use list comprehensions now that dart supports them.
any time a map block ends with .toList(), list comprehensions are probably the better option.